src/components/MovieForm/MovieForm.tsx

```diff
@@ -0,0 +1,152 @@
+import styles from "../MovieForm/MovieForm.module.css";
+import React from "react";
+import {MovieDetailsData} from "../../types/MovieDetailsData";
+
+interface MovieFormProps {
+    initialMovieInfo?: MovieDetailsData,
+    handleSubmit: (movieData: MovieDetailsData) => void;
+}
+
+const MovieForm: React.FC<MovieFormProps> = ({ initialMovieInfo, handleSubmit }) => {
+    return (
+        <form className={styles.movieForm}
+              onSubmit={(e) => {
+                  e.preventDefault();
+                  const form = e.target as HTMLFormElement;
+                  const formData = new FormData(form);
+                  const formDataObject = Object.fromEntries(formData) as Record<string, string>;
+
+                  const genres = formData.getAll("genres") as string[];
+
+                  const id = initialMovieInfo?.id ?? Math.floor(Math.random() * 10000);
+
+                  const movieData = {
+                      id,
+                      imageUrl: initialMovieInfo?.imageUrl || "",
+                      title: formDataObject.title,
+                      releaseDate: formDataObject.releaseDate,
+                      movieUrl: formDataObject.movieUrl,
+                      rating: formDataObject.rating,
+                      genres,
+                      runtime: formDataObject.runtime,
+                      description: formDataObject.description,
+                  };
+
+                  handleSubmit(movieData);
+              }
+            }
```

Comment:
move it to separate function 
![image](https://github.com/user-attachments/assets/783c4334-d733-489e-92b7-72ead580c600)

because it's difficult to see markup